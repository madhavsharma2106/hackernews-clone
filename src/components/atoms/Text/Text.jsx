import React from "react";
import classNames from "classnames";

import "./Text.scss";

export const TextVariant = {
  regular: "regular-text",
  subText: "sub-text",
  link: "link-text",
};

function Text({ variant = TextVariant.regular, children, style }) {
  const classes = classNames([`text-${variant}`]);

  return (
    <p style={style} className={classes}>
      {children}
    </p>
  );
}

export default Text;
