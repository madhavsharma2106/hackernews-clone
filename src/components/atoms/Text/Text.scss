@import "../../styles/main.scss";

.regular-text {
  font-size: $font-200;
  color: $primary-black;
}

.white-heading-text {
  font-size: $font-200;
  color: $white;
  font-weight: bold;
}

.link-text {
  font-size: $font-200;
  color: $orange;
  font-weight: bold;
}

.sub-text {
  font-size: $font-100;
  color: $grey;
}

.inline-text {
  display: inline;
}

.sub-text-dark-text {
  font-size: $font-100;
  color: $primary-black;
}
