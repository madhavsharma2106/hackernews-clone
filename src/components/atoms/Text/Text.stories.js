import React from "react";
import Text, { TextVariant } from "./Text";
import { withKnobs, text, select } from "@storybook/addon-knobs";
import { withInfo } from "@storybook/addon-info";
import { includeVariants } from "../../../utils";

export default {
  component: Text,
  title: "Atoms/Text",
  decorators: [withKnobs, withInfo],
  parameters: {
    info: { inline: true, header: false, source: true },
  },
};

export const Sample = () => {
  const sample = text("text", "Sample Text");
  const variant = select(
    "Variant",
    includeVariants(TextVariant),
    TextVariant.regular
  );

  return <Text variant={variant}>{sample}</Text>;
};

export const ClickableText = () => {
  const sample = text("text", "This Text is clickable");
  const variant = select(
    "Variant",
    includeVariants(TextVariant),
    TextVariant.regular
  );

  return (
    <Text variant={variant} clickable>
      {sample}
    </Text>
  );
};

export const InlineText = () => {
  const sample = text("text", "This Text is rendered as Display:Inline");
  const variant = select(
    "Variant",
    includeVariants(TextVariant),
    TextVariant.regular
  );

  return (
    <Text variant={variant} inline>
      {sample}
    </Text>
  );
};
