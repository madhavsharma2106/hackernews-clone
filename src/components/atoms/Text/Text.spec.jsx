import { shallow } from "enzyme";
import React from "react";
import Text, { TextVariant } from "./Text";

describe("Text Component", () => {
  it("should render with default properties", () => {
    const wrapper = shallow(<Text></Text>);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with variants ", () => {
    const wrapper = shallow(<Text variant={TextVariant.subText}> Test</Text>);
    expect(wrapper).toMatchSnapshot();
  });
});
