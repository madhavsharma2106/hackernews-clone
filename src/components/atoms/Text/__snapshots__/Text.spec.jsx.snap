// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should render with default properties 1`] = `
<p
  className="regular-text"
/>
`;

exports[`Text Component should render with variants  1`] = `
<p
  className="sub-text"
>
   Test
</p>
`;
